src/software/sensor_fusion/filter/ball_filter.h

```diff
@@ -80,65 +118,70 @@ class BallFilter
-     * @param field The field being played on.
-     */
-    void addNewDetectionsToBuffer(std::vector<BallDetection> new_ball_detections,
-                                  const Field &field);
-
-    /**
-     * Estimates the ball's velocitybased on the current detections in the given buffer.
-     * If the ball_regression_line is provided, the detection positions are projected onto
-     * the line before the velocities are calculated. If no velocity can be estimated,
-     * std::nullopt is returned.
+     * @throws std::invalid_argument if ball_detections has less than 2 elements
      *
-     * @param ball_detections The ball detections to use to calculate
-     * @param ball_regression_line The ball_regression_line to snap detections to before
-     * calculating velocities.
-     * @return A struct containing various estimates of the ball's velocity based on the
-     * given detections. If no velocity can be estimated, std::nullopt is returned
+     * @param ball_detections The ball detections to fit
+     *
+     * @return The line of best fit through the given ball detection positions
      */
-    std::optional<BallVelocityEstimate> estimateBallVelocity(
-        boost::circular_buffer<BallDetection> ball_detections,
-        const std::optional<Line> &ball_regression_line = std::nullopt);
+    static Line calculateLineOfBestFit(
+        boost::circular_buffer<BallDetection> ball_detections);
 
     /**
      * Given a list of ball detections, use linear regression to find a line of best fit
      * through the ball positions, and calculate the error of this regression.
      *
+     * @throws std::invalid_argument if ball_detections has less than 2 elements
+     *
      * @param ball_detections The ball detections to use in the regression
+     *
      * @return A struct containing the regression line and error of the linear regression
      */
-    LinearRegressionResults getLinearRegressionLine(
+    static LinearRegressionResults calculateLinearRegression(
         boost::circular_buffer<BallDetection> ball_detections);
 
     /**
-     * Uses linear regression to filter the given list of ball detections to fine the
-     * current "real" state of the ball.
+     * Returns the filtered position of the ball given a buffer of ball detections
+     * and the line of best fit through them.
      *
-     * @param ball_detections The detections to filter
-     * @return The filtered current state of the ball. If a filtered result cannot be
-     * calculated, returns std::nullopt
+     * @throws std::invalid_argument if ball_detections has less than 2 elements
+     *
+     * @param ball_detections The ball detections
+     * @param regression_line The line of best fit through the ball positions
+     *
+     * @return The filtered position of the ball
      */
-    std::optional<TimestampedBallState> estimateBallState(
-        boost::circular_buffer<BallDetection> ball_detections);
+    static Point getFilteredPosition(
```

Comment:
should this be `estimateBallPosition` to be consistent with `estimateBallVelocity`

Reply:
done